dashboard/app/models/script.rb

```diff
@@ -770,16 +770,18 @@ def self.beta?(name)
   end
 
   def get_script_level_by_id(script_level_id)
-    script_levels.find {|sl| sl.id == script_level_id.to_i}
+    script_levels.find(id: script_level_id.to_i)
   end
 
   def get_script_level_by_relative_position_and_puzzle_position(relative_position, puzzle_position, lockable)
     relative_position ||= 1
-    script_levels.to_a.find do |sl|
-      sl.lesson.lockable? == lockable &&
+    script_levels.includes(:lesson).find do |sl|
+      # make sure we are checking the native properties of the script level
```

Comment:
I thought `includes` eager loads the associations, so isn't it all in memory at this point anyway?

Reply:
Yeah, after testing it seems like either adding the includes or rearranging the order would get us the improvement we desire. I figured there's no reason not to just do both

Reply:
Oh, actually: after looking at the failing tests, it turns out one good reason to not do both is that `includes` will eager-load the relation even for those levels for which it is not necessary, which increases the expected number of queries for several of our tests that care about that. To keep this change as small as possible, I'm going to take that back out.